framework/transport/ieee1905_transport/include/mapf/transport/ieee1905_transport_messages.h

```diff
@@ -127,49 +137,108 @@ class CmduXxMessage : public Message {
 
         return os << ss.str();
     }
+};
 
-protected:
-    static const std::string build_topic(const std::string prefix, uint16_t ether_type,
-                                         uint16_t msg_type)
+class CmduRxMessage : public CmduXxMessage {
+public:
+    explicit CmduRxMessage(std::initializer_list<Frame> frame = {})
+        : CmduXxMessage(uint32_t(Type::CmduRxMessage), frame)
     {
-        std::ostringstream topic;
-        topic << prefix << std::setfill('0') << std::setw(4) << std::hex << ether_type << "."
-              << std::setfill('0') << std::setw(4) << std::hex << msg_type;
-        return topic.str();
     }
 };
 
-class CmduRxMessage : public CmduXxMessage {
-    using CmduXxMessage::CmduXxMessage; // inherit base class constructors
+class CmduTxMessage : public CmduXxMessage {
+public:
+    explicit CmduTxMessage(std::initializer_list<Frame> frame = {})
+        : CmduXxMessage(uint32_t(Type::CmduTxMessage), frame)
+    {
+    }
+};
+
+class SubscribeMessage : public Message {
+    static const uint8_t kVersion = 0;
 
 public:
-    static const std::string kTopicPrefix;
+    /**
+     * Maximal number of types for a single subscribe/unsubscribe message
+     */
+    static constexpr int MAX_SUBSCRIBE_TYPES = 64;
+
+    /**
+     * The type of the request
+     */
+    enum class ReqType : uint8_t {
+        INVALID     = 0,
+        SUBSCRIBE   = 1,
+        UNSUBSCRIBE = 2,
+    };
 
-    virtual const std::string topic_prefix() const override { return kTopicPrefix; }
+    /**
+     * Message type to subscribe/unsubscribe
+     */
+    union MsgType {
+        struct {
+            uint32_t internal : 1;        // 0 - external [cmdu], 1 - internal
```

Comment:
What is internal used for? the comment is unclear - we are using CMDUs also for internal communication.

Reply:
This bitfield is used to create subscribe/unsubscribe opcodes for the broker.
The `internal` bit specifies whether it's an internal (transport) message or an external (cmdu) message.